apps/openmw/mwmechanics/aiescort.cpp

```diff
@@ -23,10 +23,11 @@
 
 namespace MWMechanics
 {
-    AiEscort::AiEscort(const std::string &actorId, int duration, float x, float y, float z)
+    AiEscort::AiEscort(const std::string &actorId, int duration, float x, float y, float z, bool repeat)
     : mActorId(actorId), mX(x), mY(y), mZ(z), mRemainingDuration(static_cast<float>(duration))
     , mCellX(std::numeric_limits<int>::max())
     , mCellY(std::numeric_limits<int>::max())
+    , mRepeat(repeat)

```

Comment:
Instead of adding an 'mRepeat' variable to each subclass, you should probably put it in the AiPackage class and initialize it using the AiPackage constructor. Then you wouldn't need to duplicate the getRepeat function in each subclass.